vyos.configdict(): T4369: leaf_node_changed() must return True when node is added

c-po opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change Summary
<!--- Provide a general summary of your changes in the Title above -->
vyos.configdict(): T4369: leaf_node_changed() must return True when node is added

(cherry picked from commit c685c0f762ea054c7a220bde625fdab549bbbdd2)

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- All submitted PRs must be linked to a Task on Phabricator. -->
* https://phabricator.vyos.net/T4369

## Component(s) name
<!-- A rather incomplete list of components: ethernet, wireguard, bgp, mpls, ldp, l2tp, dhcp ... -->

vyos.configdict

## Proposed changes
<!--- Describe your changes in detail -->
Added missing branch in if clause

## How to test
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
like this
```
-->

OpenVPN daemon must restart itself once `openvpn-option` is added to the CLI the first time.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

c-po committed:
vyos.configdict(): T4369: leaf_node_changed() must return True when node is added

(cherry picked from commit c685c0f762ea054c7a220bde625fdab549bbbdd2)

jestabro commented:
Pausing this approval while checking configtests ...

jestabro reviewed (commented):
Checking possible error in configtests.

c-po committed:
vyos.configdict(): T4369: bugfix - execution order in leaf_node_changed()

Commit c685c0f7 ("vyos.configdict(): T4369: leaf_node_changed() must return True
when node is added") added a code path then a node was newly added to the CLI.
Unfortunately it turned out that this introduced a regression:

File "/usr/lib/python3/dist-packages/vyos/ifconfig/wireguard.py", line 230, in update
  super().update(config)
File "/usr/lib/python3/dist-packages/vyos/ifconfig/interface.py", line 1428, in update
  for addr in list_diff(config['address_old'], new_addr):
File "/usr/lib/python3/dist-packages/vyos/configdict.py", line 105, in list_diff
  return [item for item in first if item not in second]
TypeError: 'bool' object is not iterable

The execution order of the if statements is essential and the new check was
moved to the bottom to not interfere with the existing logic.

(cherry picked from commit 1a1094c28e32c3d6d072cf14a38aa631d51b8aee)
